packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1530,31 +1530,16 @@ export function getDefaultService(program: Program): Service | undefined {
   }
   return services[0];
 }
-
 /**
- * Get the default api-version both from versioned and service decorator
- * TODO: remember to switch to TCGC once the fix is done
- * @param program
- * @param dpgContext
- * @returns default api-version value
+ * Return the default api version from the program; undefined if no default
  */
-export function getEnrichedDefaultApiVersion(
+export function getDefaultApiVersionString(
   program: Program,
   dpgContext: SdkContext
 ): string | undefined {
-  const serviceNamespace = getDefaultService(program);
-  if (!serviceNamespace) {
-    return;
-  }
-
-  const defaultVersion = getDefaultApiVersion(
-    dpgContext,
-    serviceNamespace!.type
-  );
-  if (defaultVersion) {
-    return defaultVersion.value;
-  }
-  return serviceNamespace.version;
+  return getDefaultService(program)
+    ? getDefaultApiVersion(dpgContext, getDefaultService(program)!.type)?.value
+    : undefined;
```

Comment:
Get rid of the logic: extract default apiVersion from @service because this version is deprecated.